src/jansson.h

```diff
@@ -242,6 +242,30 @@ int json_object_iter_set(json_t *object, void *iter, json_t *value)
     return json_object_iter_set_new(object, iter, json_incref(value));
 }
 
+static JSON_INLINE
+int json_object_update_new(json_t *object, json_t *other)
+{
+    int ret = json_object_update(object, other);
+    json_decref(other);
```

Comment:
I'm not entirely sure, but I think that you should call `json_decref` here only if `ret == 0`. Otherwise the `other` object will be freed if the `json_object_update` call fails. Could you check the semantics of other already existing `_new` functions to align the reference counting semantics with them?

Reply:
The same applies to the other two added functions also.

Reply:
I think no matter the ```json_object_update``` call succeed or not, the ```other``` always need to be freed, since the reference of ```other``` will be stolen by this ```json_object_update_new``` function.
In fact, this function will be called like:
```C
json_object_update_new(object, json_pack("{s{ss}}", "foo", "bar", "baz"));
```

Here ```other```, created by calling ```json_pack```, is anonymous, which means it cannot be freed outside the ```json_object_update_new```

Reply:
Ah, you're right.